<center><h1> <span style='color:#fbec5d'>Objective</span></h1></center>

<center><i style='color:orange'> This project was created by Ruth Ovuike</i></center>

The objective is to crate a recommendation system using the Collaborative based method by using the Machine Learning Method called Nearest Neighbours. Nearest Neighbour Analysis measures the spread or distribution of something over a geographical space. It provides a numerical value that describes the extent to which a set of points are clustered or uniformly spaced.  Cosine similarity will be used to determine similarity between vectors.

<p style='font-weight:bold'>This <span style='color:#fbec5d'>libary 'Pandas'</span> is uesd to read the data from CSV files and compute it into a dataframe.</p>

In [6]:
# importing module
import pandas as pd

<p style='font-weight:bold'><span style='color:#fbec5d'>Matplotlib</span> is a libary that will be used in this analysis for creating <span style='color:#fbec5d'>visualizations</span>.</p>

In [1]:
# importing module
import matplotlib.pyplot as plt

<p style='font-weight:bold'>The <span style='color:#fbec5d'>Collections</span> is a built in <span style='color:#fbec5d'>Python module</span>, in this analysis it will be used to detect repatiton in a list.</p>

In [2]:
# importing module
from collections import Counter

<p style='font-weight:bold'>This library <span style='color:#fbec5d'>'Numpy'</span> is a <span style='color:#fbec5d'>Python libary</span> that will be used in this analysis to create arrays.</p>

In [3]:
# importing module
import numpy as np

<p style='font-weight:bold'>This module <span style='color:#fbec5d'>'CountryInfo'</span> is a <span style='color:#fbec5d'>Python module</span> for returning <span style='color:#fbec5d'>Data</span> about <span style='color:#fbec5d'>Countries, ISO info and States/Provinces</span> within them.</p>

In [4]:
from countryinfo import CountryInfo

<p style='font-weight:bold'><span style='color:#fbec5d'>Dataframe_image</span> is a module that will be used in this analysis to save dataframes as pictures.</p>

In [5]:
# importing module
import dataframe_image as dfi

In [6]:
import math

In [7]:
from pandas.plotting import table

In [8]:
# helps create highcharts graphs
from pandas_highcharts.core import serialize
from pandas_highcharts.display import display_charts
import json

<p style='font-weight:bold'>Reading the data from the <span style='color:#fbec5d'>'BX-Users.csv'</span> csv file carries <span style='color:#fbec5d'> the information about Movies</span> and made into a dataframe. This dataframe will be  called <span style='color:#fbec5d'>class_info</span>.</p>

In [833]:
%store region1
%store region11
%store region
%store subregion
%store df
%store df1
%store df2

Stored 'region1' (list)
Stored 'region11' (list)
Stored 'region' (list)
Stored 'subregion' (list)
Stored 'df' (DataFrame)
Stored 'df1' (DataFrame)
Stored 'df2' (DataFrame)


In [9]:
%store -r region1
%store -r region11
%store -r region
%store -r subregion
%store -r df
%store -r df1
%store -r df2

In [7]:
### The data frame is called house_info
class_info = pd.read_csv("BX-Users.csv", on_bad_lines='skip', encoding='cp1252',names=['Col1','Col2','Col3'])#error_bad_lines=False,
pd.set_option('display.max_columns', None)

In [8]:
class_info = class_info.drop(0)

In [9]:
class_info = class_info.reset_index()

In [10]:
class_info = class_info.drop('index', axis=1)

<p style='font-weight:bold'>The <span style='color:#fbec5d'>'class_info'</span> dataframe.</p>

In [11]:
class_info

Col1              Col2                   Col3
0                  1;"nyc          new york              usa";NULL
1             2;"stockton        california              usa";"18"
2               3;"moscow   yukon territory           russia";NULL
3                4;"porto          v.n.gaia         portugal";"17"
4          5;"farnborough             hants   united kingdom";NULL
...                   ...               ...                    ...
276266   278854;"portland            oregon              usa";NULL
276267     278855;"tacoma        washington   united kingdom";"50"
276268   278856;"brampton           ontario           canada";NULL
276269  278857;"knoxville         tennessee              usa";NULL
276270     278858;"dublin               n/a          ireland";NULL

[276271 rows x 3 columns]

In [12]:
type(class_info.Col1[0])

str

In [693]:
user,location1 =[],[]
for i in class_info.Col1:
    #i = i.split(";")
    if isinstance(i, float):
            user.append(i)
            location1.append('Unknown')
    if isinstance(i, str):
        i = i.split(";")
        user.append(int(str(i[0])))
        location1.append(i[1][1:])

In [13]:
age,location2 =[],[]
for i in class_info.Col3:
    if i == '':
        age.append('Unknown')
        location2.append('Unknown')
    if isinstance(i, float):
        age.append(i)
        location2.append('Unknown')
    if isinstance(i, str):    
        i = i.split(";")
        if len(i) == 1:
            location2.append(i)
            age.append('Unknown')
        elif len(i) == 2:
            location2.append(i[0][:-1])
            age.append(i[1])
        elif len(i) == 3:
            age.append(i[0][:-1])
            location2.append(i[1:])
        else: 
            age.append(i[-1])
            location2.append('Unknown')

In [14]:
new_location1 = []
for i in location2:
    if isinstance(i, list):
        if len(i) == 1:
                new_location1.append(i[0].strip().replace('"', ''))
        if len(i) == 2:
                new_location1.append(i[1].strip().replace('"', ''))
    if isinstance(i, str):
                new_location1.append(i.strip().replace('"', ''))

In [15]:
filtered_location2 = []
for i in new_location1: 
    if i == '' or i == 'n/a' or i == '-------' or i == 'na' or len(i) == 1 or i == 'öð¹ú' or i == '19104'\
    or i == 'v9v 1h4' or i == 'sri lanka\\n/a\\'or i == 'mã?â©xico' or i == '£ódÿ' or i == '?ú?{'\
    or i == 'lkjlj' or i == 'espaã?â±a' or i == 'n/a - on the road' or i == 'none':
        filtered_location2.append('Unknown')
    elif i == 'Unknown':filtered_location2.append(i)
    else: filtered_location2.append(i.title())

In [16]:
filtered_location1 = []
for i in location1: 
    if i == '' or i == 'n/a' or i == '-------' or i == 'na' or len(i) == 1 or i == 'öð¹ú' or i == '19104'\
    or i == 'v9v 1h4' or i == 'sri lanka\\n/a\\'or i == 'mã?â©xico' or i == '£ódÿ' or i == '?ú?{'\
    or i == 'lkjlj' or i == 'espaã?â±a' or i == 'n/a - on the road' or i == 'none':
        filtered_location1.append('Unknown')
    elif i == 'Unknown':filtered_location1.append(i)
    else: filtered_location1.append(i.title())

In [17]:
filtered_location3 = []
for i in class_info.Col2:
    if isinstance(i, float):
        filtered_location3.append('Unknown')
    elif isinstance(i, int):
        filtered_location3.append('Unknown')
    elif i == '' or i == 'n/a' or i == '-------' or i == 'na' or len(i) == 1 or i == 'öð¹ú' or i == '19104'\
    or i == 'v9v 1h4' or i == 'sri lanka\\n/a\\'or i == 'mã?â©xico' or i == '£ódÿ' or i == '?ú?{'\
    or i == 'lkjlj' or i == 'espaã?â±a' or i == 'n/a - on the road' or i == 'none' or i == ' N/A':
        filtered_location3.append('Unknown')
    elif i == 'Unknown':filtered_location3.append(i)
    else:
        filtered_location3.append(i.title().strip())

In [18]:
all_location = []
for i in range(276271):
    all_location.append(filtered_location1[i]+', '+filtered_location3[i]+', '+filtered_location2[i])

In [19]:
age1 = []
for i in age:
    if i == 'nan':age1.append('Unknown')
    else:
        age1.append(i)

In [688]:
filtered_age = []
for i in age1:
    if i == 'NULL' or i == '"0"' or i =='0"' or i =='':filtered_age.append('Unknown')
    elif isinstance(i, float):filtered_age.append(i)
    elif i == 'nan':filtered_age.append('Unknown')
    elif i == 'Unknown':filtered_age.append(i)
    else:
        filtered_age.append(int(str(i.replace('"',''))))

In [588]:
filterr = []
for i in Counter(filtered_location2):
    if i == 'Unknown' or i == '1407 København K' or i == 'Somewherein Space' or i =='Temple'\
        or i == '' or i == 'Csa' or i == 'Heard Of It? :O)' or i == 'Space' or i == 'Duck'\
        or i == 'You Listed Stroud!)' or i == 'Bruxelles' or i == 'De' or i == 'Doodedoo' or i == 'Spring'\
        or i == 'Spencer' or i == 'Baxter' or i == 'Evil Empire' or i == 'Bbbzzzzz' or i == 'Rice'\
        or i == '\\N/A\\' or i == '13 - 10ºesq.' or i == '23232'or i == '61 Men' or i =='Heard Of It? :O)':
        filterr.append(' ')
    else:filterr.append(i)

In [589]:
len(filterr)

2433

In [595]:
#region1 = []
geolocator = Nominatim(user_agent="geoapiExercises")
for i in range(276271):
    if filtered_location2[i] == 'Unknown' or filtered_location2[i] == '1407 København K' or \
    filtered_location2[i] == 'Somewherein Space' or filtered_location2[i] =='Temple'\
    or filtered_location2[i] == '' or filtered_location2[i] == 'Csa' or filtered_location2[i] == 'Heard Of It? :O)' \
    or filtered_location2[i] == 'Space' or filtered_location2[i] == 'Duck'\
    or filtered_location2[i] == 'You Listed Stroud!)' or filtered_location2[i] == 'Bruxelles'\
    or filtered_location2[i] == 'De' or filtered_location2[i] == 'Doodedoo' or filtered_location2[i] == 'Spring'\
    or filtered_location2[i] == 'Spencer' or filtered_location2[i] == 'Baxter' or filtered_location2[i] == \
    'Evil Empire' or filtered_location2[i] == 'Bbbzzzzz' or filtered_location2[i] == 'Rice'\
    or filtered_location2[i] == '\\N/A\\' or filtered_location2[i] == '13 - 10ºesq.' or filtered_location2[i]\
    == '23232'or filtered_location2[i] == '61 Men':
        region1.append('Unknown')
    elif filtered_location2[i] == ' ':region1.append((i,filtered_location2[i],'Unknown'))
    elif filtered_location2[i] == 'Kazakhstan':region1.append((i,filtered_location2[i],'Kazakhstan'))
    elif filtered_location2[i] == 'Selangor D.E.':region1.append((i,filtered_location2[i],'Peninsular Malaysia'))
    elif filtered_location2[i] == 'Saint Loius':region1.append((i,filtered_location2[i],'Missouri'))
    elif filtered_location2[i] == 'La Chine Éternelle !':region1.append((i,filtered_location2[i],'Romania'))
    elif filtered_location2[i] == 'K. Mozkin':region1.append((i,filtered_location2[i],'Israel'))
    elif filtered_location2[i] == 'La Chine Éternelle!':region1.append((i,filtered_location2[i],'China'))
    elif filtered_location2[i] == 'Sth Island N>Z>':region1.append((i,filtered_location2[i],'New Zealand'))
    elif filtered_location2[i] == 'Dfg':region1.append((i,filtered_location2[i],'Germany'))
    elif filtered_location2[i] == 'Êàëóæñêàÿ Îáë':region1.append((i,filtered_location2[i],'Russia'))
    elif filtered_location2[i] == 'Bfe':region1.append((i,filtered_location2[i],'Switzerland'))
    elif filtered_location2[i] == 'U.S.A>' or filtered_location2[i] =='Good Old U.S.A.':region1.append((i,filtered_location2[i],'United States'))
    elif filtered_location2[i] == 'Region Of Capital / Bratislava':region1.append((i,filtered_location2[i],'Slovakia'))
    elif filtered_location2[i] == 'Grand Manan':region1.append((i,filtered_location2[i],'Canada'))
    elif filtered_location2[i] == 'La Chine Eternelle!':region1.append((i,filtered_location2[i],'Romania'))
    elif filtered_location2[i] == 'Address Is:Adem Ademoski 13':region1.append((i,filtered_location2[i],'Greece'))
    elif filtered_location2[i] == 'S.Corea':region1.append((i,filtered_location2[i],'South Korea'))
    elif filtered_location2[i] == 'Ussurfing':region1.append((i,filtered_location2[i],'United States'))
    elif filtered_location2[i] == 'North/West Vancouver':region1.append((i,filtered_location2[i],'Canada'))
    elif filtered_location2[i] == 'Saint Luica':region1.append((i,filtered_location2[i],'Saint Lucia'))
    elif filtered_location2[i] == 'St.Thomasi':region1.append((i,filtered_location2[i],'United States'))
    elif filtered_location2[i] == 'Land Of The Free':region1.append((i,filtered_location2[i],'United States'))
    else:
        loc = geolocator.geocode(str(filtered_location2[i]))
        if loc is None :region1.append((i,filtered_location2[i],'Unknown')) 
        else:region1.append((i,filtered_location2[i],loc.raw['display_name']))
        

GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Usa&format=json&limit=1 (Caused by ReadTimeoutError("HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Read timed out. (read timeout=1)"))

In [667]:
sep_value = []
for i in region1+region11:sep_value.append(i[2].split(", "))

new_value = []
for i in sep_value:new_value.append(i[-1].split("/"))

new_value1 = []
for i in new_value:new_value1.append(i[-1].strip())

In [669]:
new_value2 = []
for i in new_value1:
    if i == 'Россия':new_value2.append('Russia')
    elif i == '中国':new_value2.append('China')
    elif i == 'México':new_value2.append('Mexico')
    elif i == 'Italia':new_value2.append('Italy')
    elif i == 'العراق':new_value2.append('Iraq')
    elif i == 'Aotearoa':new_value2.append('New Zealand')
    elif i == 'ایران':new_value2.append('Iran')
    elif i == 'Perú':new_value2.append('Peru')
    elif i == 'Босна и Херцеговина':new_value2.append('Bosnia and Herzegovina')
    elif i == 'Việt Nam':new_value2.append('Vietnam')
    elif i == 'ශ්\u200dරී ලංකාව இலங்கை':new_value2.append('Uknown')
    elif i == 'پاکستان':new_value2.append('Pakistan')
    elif i == 'Türkiye':new_value2.append('Turkey')
    elif i == 'Ελλάς':new_value2.append('Greece')
    elif i == 'الإمارات العربية المتحدة':new_value2.append('United Arab Emirates')
    elif i == 'الكويت':new_value2.append('Kuwait')
    elif i == '臺灣':new_value2.append('Taiwan')
    elif i == 'България':new_value2.append('Bulgaria')
    elif i == 'Österreich':new_value2.append('Austria')
    elif i == 'România':new_value2.append('Romania')
    elif i == 'Shqipëria':new_value2.append('Albania')
    elif i == 'مصر':new_value2.append('Egypt')
    elif i == 'تونس':new_value2.append('Tunisia')
    elif i == 'Oʻzbekiston':new_value2.append('Uzbekistan')
    elif i == 'قطر':new_value2.append('Qatar')
    elif i == 'سوريا':new_value2.append('Syria')
    elif i == 'السعودية':new_value2.append('Saudi Arabia')
    elif i == 'السودان':new_value2.append('Sudan')
    elif i == 'ประเทศไทย':new_value2.append('Thailand')
    elif i == 'Paraguái':new_value2.append('Paraguay')
    elif i == 'Moçambique':new_value2.append('Mozambique')
    elif i == 'المغرب':new_value2.append('Morocco')
    elif i == 'افغانستان':new_value2.append('Afghanistan')
    elif i == 'ישראל':new_value2.append('Israel')
    elif i == 'الجزائر':new_value2.append('Algeria')
    elif i == 'ኢትዮጵያ':new_value2.append('Ethiopia')
    elif i == 'España':new_value2.append('Spain')
    elif i == '日本':new_value2.append('Japan')
    elif i == 'Ísland':new_value2.append('Iceland')
    elif i == 'Lëtzebuerg':new_value2.append('Luxembourg')
    elif i == 'Беларусь':new_value2.append('Belarus')
    elif i == 'الأردن':new_value2.append('Jordan')
    elif i == 'Кыргызстан':new_value2.append('Kyrgyzstan')
    elif i == 'República Dominicana':new_value2.append('Dominican Republic')
    elif i == 'বাংলাদেশ':new_value2.append('Bangladesh')
    elif i == 'البحرين':new_value2.append('Bahrain')
    elif i == 'Uruguá':new_value2.append('Uruguay')
    elif i == 'Србија':new_value2.append('Serbia')
    elif i == 'Føroyar':new_value2.append('Faroe Islands')
    elif i == 'Česko':new_value2.append('Czechia')
    elif i == '대한민국':new_value2.append('South Korea')
    elif i == 'اليمن':new_value2.append('Yemen')
    elif i == 'Κύπρος - Kıbrıs':new_value2.append('Cyprus')
    elif i == 'ኤርትራ Eritrea إرتريا':new_value2.append('Eritrea')
    elif i == 'Северна Македонија':new_value2.append('North Macedonia')
    elif i == 'Sāmoa':new_value2.append('Samoa')
    elif i == 'Civitas Vaticana - Città del Vaticano':new_value2.append('Italy')
    elif i == 'Comores Komori جزر القمر':new_value2.append('Comoros')
    elif i == 'موريتانيا':new_value2.append('Mauritania')
    elif i == 'اGuinée':new_value2.append('Guinea')
    elif i == 'Қазақстан':new_value2.append('Kazakhstan')
    elif i == 'Türkmenistan':new_value2.append('Turkmenistan')
    elif i == 'ព្រះរាជាណាចក្រ\u200bកម្ពុជា':new_value2.append('Cambodia')
    elif i == 'São Tomé e Príncipe':new_value2.append('Guinea')
    elif i == 'République démocratique du Congo':new_value2.append('Democratic Republic of the Congo')
    elif i == 'Djibouti جيبوتي':new_value2.append('Djibouti')
    elif i == 'تشاد':new_value2.append('Chad')
    elif i == 'Ascension and Tristan da Cunha':new_value2.append('Saint Helena')
    elif i == 'ليبيا':new_value2.append('Libya')
    elif i == 'Bénin':new_value2.append('Benin')
    elif i == '조선민주주의인민공화국':new_value2.append('North Korea')
    elif i == 'Ōu':new_value2.append('Romania')
    elif i == 'ປະເທດລາວ':new_value2.append('Laos')                      
    elif i == 'Sénégal':new_value2.append('Senegal')              
    elif i == 'Монгол улс ᠮᠤᠩᠭᠤᠯ ᠤᠯᠤᠰ':new_value2.append('Mongolia')        
    elif i == 'Panamá':new_value2.append('Panama')
    elif i == 'إرتريا':new_value2.append('Eritrea')
    elif i == 'الصومال':new_value2.append('Somalia')
    elif i == 'မြန်မာ':new_value2.append('Myanmar')
    elif i == 'Հայաստան':new_value2.append('Armenia')
    elif i == 'Тоҷикистон':new_value2.append('Tajikistan')
    elif i == 'नेपाल':new_value2.append('Nepal')
    elif i == 'لبنان':new_value2.append('Lebanon')
    elif i == 'Côte d’Ivoire':new_value2.append('Ivory Coast')
    elif i == 'ދިވެހިރާއްޖެ':new_value2.append('Maldives')
    elif i == 'عمان':new_value2.append('Oman')
    elif i == "Kūki 'Āirani":new_value2.append('Cook Islands')
    elif i == 'འབྲུགཡུལ':new_value2.append('Bhutan')
    elif i == 'Guiné-Bissau':new_value2.append('Guinea-Bissau')
    elif i == 'Magyarország':new_value2.append('Hungary')
    elif i == 'Soomaaliya الصومال':new_value2.append('Somalia')
    elif i == 'Tchad تشاد':new_value2.append('Chad')
    elif i == 'འབྲུགཡུལ་':new_value2.append('Bhutan')
    elif i == 'eSwatini':new_value2.append('Eswatini')
    elif i == 'Guinée':new_value2.append('Guinea')
    elif i == 'Україна':new_value2.append('Ukraine')
    elif i == 'Црна Гора':new_value2.append('Montenegro')
    elif i == 'not working':new_value2.append('Unknown')
    elif i == 'k':new_value2.append('Unknown')
    elif i == 'Deutschland':new_value2.append('Germany')
    elif i == 'Slovenija':new_value2.append('Slovenia')
    elif i == 'Sverige':new_value2.append('Sweden')
    elif i == 'Polska':new_value2.append('Poland')
    elif i == 'Slovensko':new_value2.append('Slovakia')
    elif i == 'British Virgin Islands':new_value2.append('United States')
    elif i == 'Gambia':new_value2.append('The Gambia')
    elif i == 'Latvija':new_value2.append('Latvia')
    elif i == 'Cabo Verde':new_value2.append('Senegal')
    elif i == 'Lietuva':new_value2.append('Lithuania')
    elif i == 'Andorra':new_value2.append('France')
    elif i == 'Cameroun':new_value2.append('Cameroon')
    elif i == 'Azərbaycan':new_value2.append('Azerbaijan')
    elif i == 'Caribbean Sea':new_value2.append('United States')
    elif i == 'Kosovo':new_value2.append('Serbia')
    elif i == 'Peninsular Malaysia':new_value2.append('Malaysia')
    elif i == 'Belau':new_value2.append('Palau')
    elif i == 'Micronesia':new_value2.append('Philippines')
    elif i == 'Papua Niugini':new_value2.append('Papua New Guinea')
    elif i == 'Maurice':new_value2.append('Mauritius')
    elif i == 'Ayiti':new_value2.append('Haiti')
    elif i == 'Missouri':new_value2.append('United States')
    elif i == 'Dhekelia':new_value2.append('Cyprus')
    elif i == 'Africa':new_value2.append('Nigeria') 
    elif i == 'Palestinian Territory':new_value2.append('Israel')
    elif i == 'Indian Ocean':new_value2.append('India')
    elif i == 'Warszawa':new_value2.append('Poland')
    elif i == 'Eswatini':new_value2.append('Mozambique')
    else:new_value2.append(i)

In [671]:
len(Counter(new_value2))

189

In [677]:
#subregion = []
for i in new_value2:
    if i == 'Unknown' or i == 'Uknown' :subregion.append('Unknown')
    elif i == 'Antarctica' :subregion.append('Antarctica')
    else:
        if i == 'Czechia': 
            i = 'Czech Republic';v = CountryInfo(i);x = v.subregion();subregion.append(x)
        elif i == 'North Macedonia': 
            i = 'Greece';v = CountryInfo(i);x = v.subregion();subregion.append(x)
        elif i == 'Montenegro': 
            i = 'Serbia';v = CountryInfo(i);x = v.subregion();subregion.append(x)
        elif i == 'Myanmar': 
            i = 'India';v = CountryInfo(i);x = v.subregion();subregion.append(x)
        elif i == 'Eswatini': 
            i = 'Mozambique';v = CountryInfo(i);x = v.subregion();subregion.append(x)
        else:v = CountryInfo(i);x = v.subregion();subregion.append(x)

In [678]:
len(subregion)

276271

In [679]:
Counter(subregion)

Counter({'Northern America': 161493,
         'Eastern Europe': 1749,
         'Southern Europe': 29089,
         'Northern Europe': 23214,
         'Australia and New Zealand': 14875,
         'South America': 3795,
         'Unknown': 2342,
         'Western Europe': 27629,
         'Central America': 1190,
         'Eastern Asia': 2353,
         'Western Asia': 1213,
         'Southern Asia': 2273,
         'Western Africa': 386,
         'South-Eastern Asia': 3356,
         'Northern Africa': 232,
         'Central Asia': 46,
         'Eastern Africa': 147,
         'Southern Africa': 346,
         'Caribbean': 445,
         'Middle Africa': 29,
         'Polynesia': 10,
         'Antarctica': 27,
         'Micronesia': 5,
         'Melanesia': 27})

In [680]:
#region = []
for i in subregion:
    if i == 'Eastern Europe' or i == 'Southern Europe' or i == 'Northern Europe' or i == 'Western Europe':
        region.append('Europe')
    elif i == 'Eastern Asia' or i == 'Southern Asia' or i == 'Northern Asia' or i == 'Western Asia'\
    or i == 'South-Eastern Asia' or i == 'Central Asia':
        region.append('Asia')
    elif i == 'Eastern Africa' or i == 'Southern Africa' or i == 'Northern Africa' or i == 'Western Africa'\
    or i == 'Middle Africa':
        region.append('Africa')
    elif i == 'Central America':region.append('Northern America')
    elif i == 'South America':region.append('Southern America')
    elif i == 'Australia and New Zealand':region.append('Oceania')
    elif i == 'Caribbean':region.append('Northern America')
    elif i == 'Polynesia':region.append('Oceania')
    elif i == 'Micronesia':region.append('Oceania')
    elif i == 'Melanesia':region.append('Oceania')
    else:region.append(i)

In [681]:
Counter(region)

Counter({'Northern America': 163128,
         'Europe': 81681,
         'Oceania': 14917,
         'Southern America': 3795,
         'Unknown': 2342,
         'Asia': 9241,
         'Africa': 1140,
         'Antarctica': 27})

In [195]:
n = []#9015
for x,i in enumerate(user):
    if isinstance(i, int) == False:n.append((x,i))

In [305]:
n

[(272528, nan)]

In [714]:
user_new = []
for i in user:
    if isinstance(i, int): user_new.append(str(int(i)))
    else: user_new.append(i)

In [719]:
df = pd.DataFrame({'User ID':user_new,'Age':filtered_age,'City/Town':filtered_location1,\
                   'Country':new_value2,'Continent':region})

In [720]:
df

User ID      Age    City/Town         Country         Continent
0            1  Unknown          Nyc   United States  Northern America
1            2       18     Stockton   United States  Northern America
2            3  Unknown       Moscow          Russia            Europe
3            4       17        Porto        Portugal            Europe
4            5  Unknown  Farnborough  United Kingdom            Europe
...        ...      ...          ...             ...               ...
276266  278854  Unknown     Portland   United States  Northern America
276267  278855       50       Tacoma  United Kingdom            Europe
276268  278856  Unknown     Brampton          Canada  Northern America
276269  278857  Unknown    Knoxville   United States  Northern America
276270  278858  Unknown       Dublin         Ireland            Europe

[276271 rows x 5 columns]

<p style='font-weight:bold'>Reading the data from the <span style='color:#fbec5d'>'BX-Books.csv'</span> csv file carries <span style='color:#fbec5d'> the information about Movies</span> and made into a dataframe. This dataframe will be  called <span style='color:#fbec5d'>class_info1</span>.</p>

In [723]:
### The data frame is called house_info
class_info1 = pd.read_csv("BX-Books.csv", on_bad_lines='skip', encoding='cp1252')#error_bad_lines=False,
pd.set_option('display.max_columns', None)

<p style='font-weight:bold'>The <span style='color:#fbec5d'>'class_info1'</span> dataframe.</p>

In [724]:
class_info1

ISBN;"Book-Title";"Book-Author";"Year-Of-Publication";"Publisher";"Image-URL-S";"Image-URL-M";"Image-URL-L"
0       0195153448;"Classical Mythology";"Mark P. O. M...                                                         
1       0002005018;"Clara Callan";"Richard Bruce Wrigh...                                                         
2       0060973129;"Decision in Normandy";"Carlo D'Est...                                                         
3       0374157065;"Flu: The Story of the Great Influe...                                                         
4       0393045218;"The Mummies of Urumchi";"E. J. W. ...                                                         
...                                                   ...                                                         
228143  0440400988;"There's a Bat in Bunk Five";"Paula...                                                         
228144  0525447644;"From One to One Hundred";"Teri Slo...                                                         
228145  006008667X;"Lily Dale : The True Story of the ...                                                         
228146  0192126040;"Republic (World's Classics)";"Plat...                                                         
228147  0767409752;"A Guided Tour of Rene Descartes' M...                                                         

[228148 rows x 1 columns]

In [743]:
class_info1.columns

Index(['ISBN;"Book-Title";"Book-Author";"Year-Of-Publication";"Publisher";"Image-URL-S";"Image-URL-M";"Image-URL-L"'], dtype='object')

In [754]:
cleaning = []
for i in class_info1['ISBN;"Book-Title";"Book-Author";"Year-Of-Publication";"Publisher";"Image-URL-S";"Image-URL-M";"Image-URL-L"']:
    i = i.replace('"','').split(';')
    cleaning.append(i[:5])

In [792]:
isbn,title,writer,year,publisher = [],[],[],[],[]
for i in cleaning:
    isbn.append(i[0])
    title.append(i[1])
    writer.append(i[2])
    publisher.append(i[4])
    try:
        if int(str(i[3])) == 0: year.append('Unknown')
        elif isinstance(int(str(i[3])), int) : year.append(i[3])
    except:
        year.append('Unknown')

In [810]:
for i,x in enumerate(title):
    if x =='Classical Mythology':print(i,x)

0 Classical Mythology
80566 Classical Mythology


In [805]:
df1 = pd.DataFrame({'ISBN':isbn,'Book-Title':title,'Book-Author':writer,\
                   'Published':year,'Publisher':publisher})

In [806]:
df1

ISBN                                         Book-Title  \
0       0195153448                                Classical Mythology   
1       0002005018                                       Clara Callan   
2       0060973129                               Decision in Normandy   
3       0374157065  Flu: The Story of the Great Influenza Pandemic...   
4       0393045218                             The Mummies of Urumchi   
...            ...                                                ...   
228143  0440400988                         There's a Bat in Bunk Five   
228144  0525447644                            From One to One Hundred   
228145  006008667X  Lily Dale : The True Story of the Town that Ta...   
228146  0192126040                        Republic (World's Classics)   
228147  0767409752  A Guided Tour of Rene Descartes' Meditations o...   

                 Book-Author Published  \
0         Mark P. O. Morford      2002   
1       Richard Bruce Wright      2001   
2               Carlo D'Este      1991   
3           Gina Bari Kolata      1999   
4            E. J. W. Barber      1999   
...                      ...       ...   
228143        Paula Danziger      1988   
228144            Teri Sloat      1991   
228145      Christine Wicker      2004   
228146                 Plato      1996   
228147   Christopher  Biffle      2000   

                                               Publisher  
0                                Oxford University Press  
1                                  HarperFlamingo Canada  
2                                        HarperPerennial  
3                                   Farrar Straus Giroux  
4                                      W. W. Norton &amp  
...                                                  ...  
228143                   Random House Childrens Pub (Mm)  
228144                                      Dutton Books  
228145                                HarperSanFrancisco  
228146                           Oxford University Press  
228147  McGraw-Hill Humanities/Social Sciences/Languages  

[228148 rows x 5 columns]

<p style='font-weight:bold'>Reading the data from the <span style='color:#fbec5d'>'BX-Books-Ratings.csv'</span> csv file carries <span style='color:#fbec5d'> the information about Movies</span> and made into a dataframe. This dataframe will be  called <span style='color:#fbec5d'>class_info</span>.</p>

In [817]:
### The data frame is called house_info
class_info2 = pd.read_csv("BX-Book-Ratings.csv", on_bad_lines='skip', encoding='cp1252')#error_bad_lines=False,
pd.set_option('display.max_columns', None)

<p style='font-weight:bold'>The <span style='color:#fbec5d'>'class_info2'</span> dataframe.</p>

In [818]:
class_info2

User-ID;"ISBN";"Book-Rating"
0            276725;"034545104X";"0"
1            276726;"0155061224";"5"
2            276727;"0446520802";"0"
3            276729;"052165615X";"3"
4            276729;"0521795028";"6"
...                              ...
1149770      276704;"1563526298";"9"
1149771      276706;"0679447156";"0"
1149772     276709;"0515107662";"10"
1149773     276721;"0590442449";"10"
1149774     276723;"05162443314";"8"

[1149775 rows x 1 columns]

In [819]:
class_info2.columns

Index(['User-ID;"ISBN";"Book-Rating"'], dtype='object')

In [820]:
cleaning2 = []
for i in class_info2['User-ID;"ISBN";"Book-Rating"']:
    i = i.replace('"','').split(';')
    cleaning2.append(i[:5])

In [827]:
user_id,isbn1,rating = [],[],[]
for i in cleaning2:
    user_id.append(i[0])
    isbn1.append(i[1])
    rating.append(i[2])

In [831]:
df2 = pd.DataFrame({'User-ID':user_id,'ISBN':isbn1,'Book-Rating':rating})

In [835]:
len(Counter(isbn1))

340551

In [836]:
len(Counter(user_id))

105281

In [832]:
df2

User-ID         ISBN Book-Rating
0        276725   034545104X           0
1        276726   0155061224           5
2        276727   0446520802           0
3        276729   052165615X           3
4        276729   0521795028           6
...         ...          ...         ...
1149770  276704   1563526298           9
1149771  276706   0679447156           0
1149772  276709   0515107662          10
1149773  276721   0590442449          10
1149774  276723  05162443314           8

[1149775 rows x 3 columns]

In [36]:
s = []
for i in df.Age:
    if isinstance(i,str):s.append(i)
    elif isinstance(i,int):s.append(i)
    else:s.append('Unknown')

In [40]:
s1 = []
for i in s:
    if isinstance(i,int) and i >= 101:s1.append('Unknown')
    else:s1.append(i)

In [44]:
category_age = []
for i in s1:
    if isinstance(i,int):
        if i == 1:category_age.append('Infant')
        elif 12 <= i <= 19 :category_age.append('Adolescence')
        elif 59 >= i >= 40 :category_age.append('Middle Aged Adult')
        elif 39 >= i >= 30 :category_age.append('Adult')
        elif 8 >= i >= 4  :category_age.append('Early Childhood')
        elif  i >= 65  :category_age.append('Old Age')
        elif 29 >= i >= 20 :category_age.append('Early Adulthood')
        elif 64 >= i >= 59 :category_age.append('Late Adulthood')
        elif 11 >= i >= 9 :category_age.append('Middle childhood')
        elif i <= 0 :category_age.append('Prenatal Development')
        elif 3 >= i >= 2 :category_age.append('Toddlerhood')
    else:category_age.append(i)

In [45]:
Counter(category_age)

Counter({'Unknown': 111420,
         'Adolescence': 18441,
         'Late Adulthood': 4423,
         'Early Adulthood': 50825,
         'Middle Aged Adult': 44756,
         'Adult': 41353,
         'Old Age': 4214,
         'Infant': 281,
         'Middle childhood': 265,
         'Toddlerhood': 144,
         'Early Childhood': 149})

In [828]:
Counter(rating)

Counter({'0': 716106,
         '5': 50974,
         '3': 5996,
         '6': 36923,
         '8': 103736,
         '7': 76457,
         '10': 78609,
         '9': 67541,
         '4': 8904,
         '1': 1770,
         '2': 2759})

In [54]:
year1 = []
for i in df1.Published:
     try:
        if int(str(i)) > 2022: year1.append('Unknown')
        else: year1.append(i)
     except:
        year1.append('Unknown')

In [ ]:
rat_category = []
for i,x in enumerate(df2.ISBN):
    for y in range(228148):
        if x == df2.ISBN[y] :rat_category.append((i,x))

In [59]:
cc

[(9561, '0195153448')]

In [13]:
image = df.head(6)

In [14]:
image1 = df1.head(6)

In [15]:
image2 = df2.head(6)

In [16]:
def highlight_cells13(x):
    df = x.copy()
    df.loc[:] = '' 
    df.iloc[:,0] = 'background-color:#FFD700;color:black;border-bottom: 2px solid black;'
    df.iloc[:,1] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'
    df.iloc[:,2] = 'background-color:#FFD700;color:black;border-bottom: 2px solid black;'
    df.iloc[:,3] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'
    df.iloc[:,4] = 'background-color:#FFD700;color:black;border-bottom: 2px solid black;'
    return df 
image1=image1.style.set_table_styles([{'selector' : '','props' : [('border','4px solid #FFD700')]},
            {"selector":"thead", 'props':[("background-color","#FAFAD2"),("color","black")]},
                                         #("font-size", "50px")]},#headinig
            #{"selector": "tr", "props": "line-height: 12px;"},
            #{"selector": "td,th", "props": "line-height: inherit; padding: 0;"},
            #{'selector':"td", "props":[("background-color","#FAFAD2"),("color"," black")]},#inside chart
            {'selector':'th.row_heading', 'props':[('background-color','#FAFAD2'),('color','black')]}
                                                  #("font-size", "50px")]},#index
             ])\
            .apply(highlight_cells13,axis=None)

In [17]:
def highlight_cells13(x):
    df = x.copy()
    df.loc[:] = '' 
    df.iloc[:,0] = 'background-color:#DAA520;color:black;border-bottom: 2px solid black;'
    df.iloc[:,1] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'
    df.iloc[:,2] = 'background-color:#DAA520;color:black;border-bottom: 2px solid black;'
    return df 
image2=image2.style.set_table_styles([{'selector' : '','props' : [('border','4px solid #DAA520')]},
            {"selector":"thead", 'props':[("background-color","#FAFAD2"),("color","black")]},
                                         #("font-size", "50px")]},#headinig
            #{"selector": "tr", "props": "line-height: 12px;"},
            #{"selector": "td,th", "props": "line-height: inherit; padding: 0;"},
            #{'selector':"td", "props":[("background-color","#FAFAD2"),("color"," black")]},#inside chart
            {'selector':'th.row_heading', 'props':[('background-color','#FAFAD2'),('color','black')]}
                                                  #("font-size", "50px")]},#index
             ])\
            .apply(highlight_cells13,axis=None)

In [18]:
def highlight_cells13(x):
    df = x.copy()
    df.loc[:] = '' 
    df.iloc[:,0] = 'background-color:#C5B358;color:black;border-bottom: 2px solid black;'
    df.iloc[:,1] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'
    df.iloc[:,2] = 'background-color:#C5B358;color:black;border-bottom: 2px solid black;'
    df.iloc[:,3] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'
    df.iloc[:,4] = 'background-color:#C5B358;color:black;border-bottom: 2px solid black;'
    #df.iloc[:,5] = 'background-color:#FAFAD2;color:black;border-bottom: 2px solid black;'

    return df 
image=image.style.set_table_styles([{'selector' : '','props' : [('border','4px solid #C5B358')]},
            {"selector":"thead", 'props':[("background-color","#FAFAD2"),("color","black")]},
                                         #("font-size", "50px")]},#headinig
            #{"selector": "tr", "props": "line-height: 12px;"},
            #{"selector": "td,th", "props": "line-height: inherit; padding: 0;"},
            #{'selector':"td", "props":[("background-color","#FAFAD2"),("color"," black")]},#inside chart
            {'selector':'th.row_heading', 'props':[('background-color','#FAFAD2'),('color','black')]}
                                                  #("font-size", "50px")]},#index
             ])\
            .apply(highlight_cells13,axis=None)

In [33]:
%%html
<script src="https://code.jquery.com/jquery-3.3.1.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<link href="https://cdn.webdatarocks.com/latest/webdatarocks.min.css" rel="stylesheet" />
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.toolbar.min.js"></script>
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.js"></script>
<script src="https://code.highcharts.com/4.2.2/highcharts.js"></script>
<script src="https://code.highcharts.com/4.2.2/highcharts-more.js"></script>
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.highcharts.js"></script>

<span class="highcharts-figure">
    <div id="containerzzz"></div>
    <p class="highcharts-description">
    </p>
</span>

In [34]:
%%js
Highcharts.chart('containerzzz', {
    chart: {
        type: 'column',
        width:880,
        height:450,
    },
    title: {
        text: '<span style="color:#996515">This is a Count Chart on Where Each User is Located in the World.<span>'
    },
    subtitle: {
        text: '<span style="color:#e5b73b">The Locatiosn is put into 7 Categories which are the Seven Continents</span>'
    },
    xAxis: {
        type: 'category',
        labels: {
            style: {
                fontSize: '13px',
                fontFamily: 'Verdana, sans-serif'
            }
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Amount of Movies'
        }
    },
    legend: {
        enabled: false
    },
    plotOptions: {
        column: {
            colorByPoint: true
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    tooltip: {
        pointFormat: 'Amount of Movies: <b>{point.y:.0f} </b>'
    },
    series: [{
        colors: ['#996515','#cba135','#e5b73b','#ffd700', '#f0e68c', '#faf0be', '#f5f5dc'],
        name: 'Bad Static Head',
        data: [
            ['Northern America', 163128],
            ['Europe', 81681],
            ['Oceania', 14917],
            ['Asia', 9241],
            ['Southern America', 3795],
            ['Unknown', 2342],
            ['Africa', 1140],
            ['Antarctica', 27]
        ]
    }]
});

<IPython.core.display.Javascript object>

In [51]:
%%html
<script src="https://code.jquery.com/jquery-3.3.1.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<link href="https://cdn.webdatarocks.com/latest/webdatarocks.min.css" rel="stylesheet" />
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.toolbar.min.js"></script>
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.js"></script>
<script src="https://code.highcharts.com/4.2.2/highcharts.js"></script>
<script src="https://code.highcharts.com/4.2.2/highcharts-more.js"></script>
<script src="https://cdn.webdatarocks.com/latest/webdatarocks.highcharts.js"></script>

<span class="highcharts-figure">
    <div id="containereee"></div>
    <p class="highcharts-description">
    </p>
</span>

In [52]:
%%js
Highcharts.chart('containereee', {
    chart: {
        type: 'bar',
        width: 400,
        height: 400
    },
    title: {
        text: 'Africa'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Africa'
    },
    xAxis: {
        categories: [ 'Northern Africa','Eastern Africa','Southern Africa','Western Africa',
                     'Central Africa'
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Africa | Users: 1,140',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:.0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#faf0be",
            data: [
                {
                    name: "Northern Africa",
                    y: 232,
                    drilldown: "Northern Africa"
                },
                {
                    name: "Southern Africa",
                    y: 346,
                    drilldown: "Southern Africa"
                },
                {
                    name: "Western Africa",
                    y: 386,
                    drilldown: "Western Africa"
                },
                {
                    name: "Eastern Africa",
                    y: 147,
                    drilldown: "Eastern Africa"
                },
                {
                    name: "Central Africa",
                    y: 29,
                    drilldown: "Central Africa"
                },
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [37]:
%%html
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://code.highcharts.com/themes/sunset.js"></script>
<script src="https://code.highcharts.com/modules/export-data.js"></script>
<script src="https://code.highcharts.com/modules/column.js"></script>

<span class="highcharts-figure">
    <div id="lucy"></div>
    <p class="highcharts-description">
    </p>
</span>

In [38]:
%%js
Highcharts.chart('lucy', {
    chart: {
        type: 'bar',
        width: 400,
        height: 400
    },
    title: {
        text: 'Asia'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Asia'
    },
    xAxis: {
        categories: ['Eastern Asia','Southern Asia','Western Asia',
                     'Central Asia'
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Asia | Users: 9,241',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#ffd700",
            data: [
                {
                    name: "Southern Asia",
                    y: 5629,
                    drilldown: "Southern Asia"
                },
                {
                    name: "Western Asia",
                    y: 1213,
                    drilldown: "Western Asia"
                },
                {
                    name: "Eastern Asia",
                    y: 2383,
                    drilldown: "Eastern Asia"
                },
                {
                    name: "Central Asia",
                    y: 46,
                    drilldown: "Central Asia"
                },
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [39]:
%%html
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://code.highcharts.com/themes/sunset.js"></script>
<script src="https://code.highcharts.com/modules/export-data.js"></script>
<script src="https://code.highcharts.com/modules/column.js"></script>

<span class="highcharts-figure">
    <div id="lucy1"></div>
    <p class="highcharts-description">
    </p>
</span>

In [40]:
%%js
Highcharts.chart('lucy1', {
    chart: {
        type: 'bar',
        width: 350,
        height: 350
    },
    title: {
        text: 'Europe'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Europe'
    },
    xAxis: {
        categories: ['Northern Europe','Eastern Europe','Southern Europe','Western Europe',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Europe | Users: 81,691',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#cba135",
            data: [
                {
                    name: "Northern Europe",
                    y: 23214,
                    drilldown: "Northern Europe"
                },
                {
                    name: "Southern Europe",
                    y: 28089,
                    drilldown: "Southern Europe"
                },
                {
                    name: "Western Europe",
                    y: 27629,
                    drilldown: "Western Europe"
                },
                {
                    name: "Eastern Europe",
                    y: 1749,
                    drilldown: "Eastern Europe"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [41]:
%%html
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://code.highcharts.com/themes/sunset.js"></script>
<script src="https://code.highcharts.com/modules/export-data.js"></script>
<script src="https://code.highcharts.com/modules/column.js"></script>

<span class="highcharts-figure">
    <div id="lucy2"></div>
    <p class="highcharts-description">
    </p>
</span>

In [42]:
%%js
Highcharts.chart('lucy2', {
    chart: {
        type: 'bar',
        width: 400,
        height: 400
    },
    title: {
        text: 'Northern America'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Northern America'
    },
    xAxis: {
        categories: ['Northern America','Central America','Caribbean Islands',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Northern America | Users: 163,128',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#996515",
            data: [
                {
                    name: "Northern America",
                    y: 161493,
                    drilldown: "Northern America"
                },
                {
                    name: "Central America",
                    y: 1190,
                    drilldown: "Central America"
                },
                {
                    name: "Caribbean Islands",
                    y: 445,
                    drilldown: "Caribbean Islands"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [43]:
%%html
<script src="https://code.highcharts.com/highcharts.js"></script>
<script src="https://code.highcharts.com/modules/exporting.js"></script>
<script src="https://code.highcharts.com/modules/accessibility.js"></script>
<script src="https://code.highcharts.com/themes/sunset.js"></script>
<script src="https://code.highcharts.com/modules/export-data.js"></script>
<script src="https://code.highcharts.com/modules/column.js"></script>

<span class="highcharts-figure">
    <div id="lucy3"></div>
    <p class="highcharts-description">
    </p>
</span>

In [44]:
%%js
Highcharts.chart('lucy3', {
    chart: {
        type: 'bar',
        width: 400,
        height: 400
    },
    title: {
        text: 'Oceania'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Oceania'
    },
    xAxis: {
        categories: ['Australia and New Zealand','Polynesia','Micronesia','Melanesia',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Oceania | Users: 14,917',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#e5b73b",
            data: [
                {
                    name: "Australia and New Zealand",
                    y: 14875,
                    drilldown: "Australia and New Zealand"
                },
                {
                    name: "Polynesia",
                    y: 10,
                    drilldown: "Polynesia"
                },
                {
                    name: "Micronesia",
                    y: 5,
                    drilldown: "Micronesia"
                },
                {
                    name: "Melanesia",
                    y: 27,
                    drilldown: "Melanesia"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

<center><h3>The Three Dataframes that wll be used in this Recommandation System</h3></center>

<center>
    <table style='border: 4px solid #FFD700;display:inline-block;margin:11px;'>
  <thead>
    <tr>
      <th class="blank level0" style='background-color: #FAFAD2;color:black'>&nbsp;</th>
      <th class="col_heading level0 col0" style='background-color: #FAFAD2;color:black; '>ISBN</th>
      <th class="col_heading level0 col1" style='background-color: #FAFAD2;color:black;'>Book-Title</th>
      <th class="col_heading level0 col2" style='background-color: #FAFAD2;color:black'>Book-Author</th>
      <th class="col_heading level0 col3" style='background-color: #FAFAD2;color:black'>Published</th>
      <th class="col_heading level0 col4" style='background-color: #FAFAD2;color:black'>Publisher</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_fc60e_level0_row0" class="row_heading level0 row0" style='background-color: #FAFAD2;color:black'>0</th>
      <td id="T_fc60e_row0_col0" class="data row0 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0195153448</td>
      <td id="T_fc60e_row0_col1" class="data row0 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Classical Mythology</td>
      <td id="T_fc60e_row0_col2" class="data row0 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Mark P. O. Morford</td>
      <td id="T_fc60e_row0_col3" class="data row0 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>2002</td>
      <td id="T_fc60e_row0_col4" class="data row0 col4" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Oxford University Press</td>
    </tr>
    <tr>
      <th id="T_fc60e_level0_row1" class="row_heading level0 row1" style='background-color: #FAFAD2;color:black'>1</th>
      <td id="T_fc60e_row1_col0" class="data row1 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0002005018</td>
      <td id="T_fc60e_row1_col1" class="data row1 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Clara Callan</td>
      <td id="T_fc60e_row1_col2" class="data row1 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Richard Bruce Wright</td>
      <td id="T_fc60e_row1_col3" class="data row1 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>2001</td>
      <td id="T_fc60e_row1_col4" class="data row1 col4" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>HarperFlamingo Canada</td>
    </tr>
    <tr>
      <th id="T_fc60e_level0_row2" class="row_heading level0 row2" style='background-color: #FAFAD2;color:black'>2</th>
      <td id="T_fc60e_row2_col0" class="data row2 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0060973129</td>
      <td id="T_fc60e_row2_col1" class="data row2 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Decision in Normandy</td>
      <td id="T_fc60e_row2_col2" class="data row2 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Carlo D'Este</td>
      <td id="T_fc60e_row2_col3" class="data row2 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>1991</td>
      <td id="T_fc60e_row2_col4" class="data row2 col4" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>HarperPerennial</td>
    </tr>
    <tr>
      <th id="T_fc60e_level0_row3" class="row_heading level0 row3" style='background-color: #FAFAD2;color:black'>3</th>
      <td id="T_fc60e_row3_col0" class="data row3 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0374157065</td>
      <td id="T_fc60e_row3_col1" class="data row3 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Flu: The Story of the Great...</td>
      <td id="T_fc60e_row3_col2" class="data row3 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Gina Bari Kolata</td>
      <td id="T_fc60e_row3_col3" class="data row3 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>1999</td>
      <td id="T_fc60e_row3_col4" class="data row3 col4" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Farrar Straus Giroux</td>
    </tr>
    <tr>
      <th id="T_fc60e_level0_row4" class="row_heading level0 row4" style='background-color: #FAFAD2;color:black'>4</th>
      <td id="T_fc60e_row4_col0" class="data row4 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0393045218</td>
      <td id="T_fc60e_row4_col1" class="data row4 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>The Mummies of Urumchi</td>
      <td id="T_fc60e_row4_col2" class="data row4 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>E. J. W. Barber</td>
      <td id="T_fc60e_row4_col3" class="data row4 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>1999</td>
      <td id="T_fc60e_row4_col4" class="data row4 col4" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>W. W. Norton &amp</td>
    </tr>
    <tr>
      <th id="T_fc60e_level0_row5" class="row_heading level0 row5" style='background-color: #FAFAD2;color:black'>5</th>
      <td id="T_fc60e_row5_col0" class="data row5 col0" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>0399135782</td>
      <td id="T_fc60e_row5_col1" class="data row5 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>The Kitchen God's Wife</td>
      <td id="T_fc60e_row5_col2" class="data row5 col2" style='background-color: #FFD700;color:black; border-bottom: 1px solid black;'>Amy Tan</td>
      <td id="T_fc60e_row5_col3" class="data row5 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>1991</td>
      <td id="T_fc60e_row5_col4" class="data row5 col4" style='background-color: #FFD700;color:black;border-bottom: 1px solid black;'>Putnam Pub Group</td>
    </tr>
  </tbody>
</table>

<table style='display:inline-block;border:4px solid #C5B358;margin:11px;'>
  <thead>
    <tr>
      <th class="blank level0" style='background-color: #FAFAD2;color:black'>&nbsp;</th>
      <th class="col_heading level0 col0" style='background-color: #FAFAD2;color:black;'>User ID</th>
      <th class="col_heading level0 col1" style='background-color: #FAFAD2;color:black;'>Age</th>
      <th class="col_heading level0 col2" style='background-color: #FAFAD2;color:black;'>City/Town</th>
      <th class="col_heading level0 col3" style='background-color: #FAFAD2;color:black;'>Country</th>
      <th class="col_heading level0 col4" style='background-color: #FAFAD2;color:black;'>Continent</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_a0b9c_level0_row0" class="row_heading level0 row0" style='background-color: #FAFAD2;color:black;'>0</th>
      <td id="T_a0b9c_row0_col0" class="data row0 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>1</td>
      <td id="T_a0b9c_row0_col1" class="data row0 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Unknown</td>
      <td id="T_a0b9c_row0_col2" class="data row0 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Nyc</td>
      <td id="T_a0b9c_row0_col3" class="data row0 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>United States</td>
      <td id="T_a0b9c_row0_col4" class="data row0 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Northern America</td>
    </tr>
    <tr>
      <th id="T_a0b9c_level0_row1" class="row_heading level0 row1" style='background-color: #FAFAD2;color:black;'>1</th>
      <td id="T_a0b9c_row1_col0" class="data row1 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>2</td>
      <td id="T_a0b9c_row1_col1" class="data row1 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>18</td>
      <td id="T_a0b9c_row1_col2" class="data row1 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Stockton</td>
      <td id="T_a0b9c_row1_col3" class="data row1 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>United States</td>
      <td id="T_a0b9c_row1_col4" class="data row1 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Northern America</td>
    </tr>
    <tr>
      <th id="T_a0b9c_level0_row2" class="row_heading level0 row2" style='background-color: #FAFAD2;color:black;'>2</th>
      <td id="T_a0b9c_row2_col0" class="data row2 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>3</td>
      <td id="T_a0b9c_row2_col1" class="data row2 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Unknown</td>
      <td id="T_a0b9c_row2_col2" class="data row2 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Moscow</td>
      <td id="T_a0b9c_row2_col3" class="data row2 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Russia</td>
      <td id="T_a0b9c_row2_col4" class="data row2 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Europe</td>
    </tr>
    <tr>
      <th id="T_a0b9c_level0_row3" class="row_heading level0 row3" style='background-color: #FAFAD2;color:black;'>3</th>
      <td id="T_a0b9c_row3_col0" class="data row3 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>4</td>
      <td id="T_a0b9c_row3_col1" class="data row3 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>17</td>
      <td id="T_a0b9c_row3_col2" class="data row3 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Porto</td>
      <td id="T_a0b9c_row3_col3" class="data row3 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Portugal</td>
      <td id="T_a0b9c_row3_col4" class="data row3 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Europe</td>
    </tr>
    <tr>
      <th id="T_a0b9c_level0_row4" class="row_heading level0 row4" style='background-color: #FAFAD2;color:black;'>4</th>
      <td id="T_a0b9c_row4_col0" class="data row4 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>5</td>
      <td id="T_a0b9c_row4_col1" class="data row4 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>Unknown</td>
      <td id="T_a0b9c_row4_col2" class="data row4 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Farnborough</td>
      <td id="T_a0b9c_row4_col3" class="data row4 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>United Kingdom</td>
      <td id="T_a0b9c_row4_col4" class="data row4 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Europe</td>
    </tr>
    <tr>
      <th id="T_a0b9c_level0_row5" class="row_heading level0 row5" style='background-color: #FAFAD2;color:black;'>5</th>
      <td id="T_a0b9c_row5_col0" class="data row5 col0" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>6</td>
      <td id="T_a0b9c_row5_col1" class="data row5 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>61</td>
      <td id="T_a0b9c_row5_col2" class="data row5 col2" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Santa Monica</td>
      <td id="T_a0b9c_row5_col3" class="data row5 col3" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>United States</td>
      <td id="T_a0b9c_row5_col4" class="data row5 col4" style='background-color: #C5B358;color:black; border-bottom: 1px solid black;'>Northern America</td>
    </tr>
  </tbody>
</table>

<table style='display:inline-block;border: 4px solid #DAA520;margin:11px;'>
  <thead>
    <tr>
      <th class="blank level0" style='background-color: #FAFAD2;color:black;'>&nbsp;</th>
      <th class="col_heading level0 col0" style='background-color: #FAFAD2;color:black; '>User-ID</th>
      <th class="col_heading level0 col1" style='background-color: #FAFAD2;color:black; '>ISBN</th>
      <th class="col_heading level0 col2" style='background-color: #FAFAD2;color:black; '>Book-Rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_3f147_level0_row0" class="row_heading level0 row0" style='background-color: #FAFAD2;color:black;'>0</th>
      <td id="T_3f147_row0_col0" class="data row0 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276725</td>
      <td id="T_3f147_row0_col1" class="data row0 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>034545104X</td>
      <td id="T_3f147_row0_col2" class="data row0 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>0</td>
    </tr>
    <tr>
      <th id="T_3f147_level0_row1" class="row_heading level0 row1" style='background-color: #FAFAD2;color:black;'>1</th>
      <td id="T_3f147_row1_col0" class="data row1 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276726</td>
      <td id="T_3f147_row1_col1" class="data row1 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>0155061224</td>
      <td id="T_3f147_row1_col2" class="data row1 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>5</td>
    </tr>
    <tr>
      <th id="T_3f147_level0_row2" class="row_heading level0 row2" style='background-color: #FAFAD2;color:black;'>2</th>
      <td id="T_3f147_row2_col0" class="data row2 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276727</td>
      <td id="T_3f147_row2_col1" class="data row2 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>0446520802</td>
      <td id="T_3f147_row2_col2" class="data row2 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>0</td>
    </tr>
    <tr>
      <th id="T_3f147_level0_row3" class="row_heading level0 row3" style='background-color: #FAFAD2;color:black;'>3</th>
      <td id="T_3f147_row3_col0" class="data row3 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276729</td>
      <td id="T_3f147_row3_col1" class="data row3 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>052165615X</td>
      <td id="T_3f147_row3_col2" class="data row3 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>3</td>
    </tr>
    <tr>
      <th id="T_3f147_level0_row4" class="row_heading level0 row4" style='background-color: #FAFAD2;color:black;' >4</th>
      <td id="T_3f147_row4_col0" class="data row4 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276729</td>
      <td id="T_3f147_row4_col1" class="data row4 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>0521795028</td>
      <td id="T_3f147_row4_col2" class="data row4 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>6</td>
    </tr>
    <tr>
      <th id="T_3f147_level0_row5" class="row_heading level0 row5" style='background-color: #FAFAD2;color:black;' >5</th>
      <td id="T_3f147_row5_col0" class="data row5 col0" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>276733</td>
      <td id="T_3f147_row5_col1" class="data row5 col1" style='background-color: #FAFAD2;color:black; border-bottom: 1px solid black;'>2080674722</td>
      <td id="T_3f147_row5_col2" class="data row5 col2" style='background-color: #DAA520;color:black; border-bottom: 1px solid black;'>0</td>
    </tr>
  </tbody>
</table>
<table>
</table></center>

<center><h3 style='color:white;'>A Count Chart of the Amount of Users in Each Continent.</h3></center>

<center><i style='color:yellow'> (the graph below is interactive, you can hover over the column charts)</i></center>

<center><div id='good1'></div></center>

<center><h3 style='color:white;'>A Count Chart of the Amount of Users in Each Coast of Each Continent.</h3></center>

<center><i style='color:yellow'> (the graph below is interactive, you can hover over the column charts)</i></center>

<table>
<tr>
    <td><div id='good2'></div><td>
    <td><div id='good3'></div><td>
    <td><div id='good4'></div><td>
    <td><div id='good5'></div><td>
    <td><div id='good6'></div><td>
    </tr>
</table>

In [45]:
%%js
Highcharts.chart('good1', {
    chart: {
        type: 'column',
        width:880,
        height:450,
    },
    title: {
        text: '<span style="color:#996515">This is a Count Chart on Where Each User is Located in the World.<span>'
    },
    subtitle: {
        text: '<span style="color:#e5b73b">The Locatiosn is put into 7 Categories which are the Seven Continents</span>'
    },
    xAxis: {
        type: 'category',
        labels: {
            style: {
                fontSize: '13px',
                fontFamily: 'Verdana, sans-serif'
            }
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Amount of Movies'
        }
    },
    legend: {
        enabled: false
    },
    plotOptions: {
        column: {
            colorByPoint: true
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    tooltip: {
        pointFormat: 'Amount of Movies: <b>{point.y:.0f} </b>'
    },
    series: [{
        colors: ['#996515','#cba135','#e5b73b','#ffd700', '#f0e68c', '#faf0be', '#f5f5dc'],
        name: 'Bad Static Head',
        data: [
            ['Northern America', 163128],
            ['Europe', 81681],
            ['Oceania', 14917],
            ['Asia', 9241],
            ['Southern America', 3795],
            ['Unknown', 2342],
            ['Africa', 1140],
            ['Antarctica', 27]
        ]
    }]
});

<IPython.core.display.Javascript object>

In [46]:
%%js
Highcharts.chart('good2', {
    chart: {
        type: 'bar',
        width: 300,
        height: 400
    },
    title: {
        text: 'Africa'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Africa'
    },
    xAxis: {
        categories: [ 'Northern Africa','Eastern Africa','Southern Africa','Western Africa',
                     'Central Africa'
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Africa | Users: 1,140',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:.0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#faf0be",
            data: [
                {
                    name: "Northern Africa",
                    y: 232,
                    drilldown: "Northern Africa"
                },
                {
                    name: "Southern Africa",
                    y: 346,
                    drilldown: "Southern Africa"
                },
                {
                    name: "Western Africa",
                    y: 386,
                    drilldown: "Western Africa"
                },
                {
                    name: "Eastern Africa",
                    y: 147,
                    drilldown: "Eastern Africa"
                },
                {
                    name: "Central Africa",
                    y: 29,
                    drilldown: "Central Africa"
                },
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [47]:
%%js
Highcharts.chart('good3', {
    chart: {
        type: 'bar',
        width: 300,
        height: 400
    },
    title: {
        text: 'Asia'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Asia'
    },
    xAxis: {
        categories: ['Eastern Asia','Southern Asia','Western Asia',
                     'Central Asia'
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Asia | Users: 9,241',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#ffd700",
            data: [
                {
                    name: "Southern Asia",
                    y: 5629,
                    drilldown: "Southern Asia"
                },
                {
                    name: "Western Asia",
                    y: 1213,
                    drilldown: "Western Asia"
                },
                {
                    name: "Eastern Asia",
                    y: 2383,
                    drilldown: "Eastern Asia"
                },
                {
                    name: "Central Asia",
                    y: 46,
                    drilldown: "Central Asia"
                },
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [48]:
%%js
Highcharts.chart('good4', {
    chart: {
        type: 'bar',
        width: 300,
        height: 400
    },
    title: {
        text: 'Europe'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Europe'
    },
    xAxis: {
        categories: ['Northern Europe','Eastern Europe','Southern Europe','Western Europe',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Europe | Users: 81,691',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#cba135",
            data: [
                {
                    name: "Northern Europe",
                    y: 23214,
                    drilldown: "Northern Europe"
                },
                {
                    name: "Southern Europe",
                    y: 28089,
                    drilldown: "Southern Europe"
                },
                {
                    name: "Western Europe",
                    y: 27629,
                    drilldown: "Western Europe"
                },
                {
                    name: "Eastern Europe",
                    y: 1749,
                    drilldown: "Eastern Europe"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [49]:
%%js
Highcharts.chart('good5', {
    chart: {
        type: 'bar',
        width: 300,
        height: 400
    },
    title: {
        text: 'Northern America'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Northern America'
    },
    xAxis: {
        categories: ['Northern America','Central America','Caribbean Islands',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Northern America | Users: 163,128',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#996515",
            data: [
                {
                    name: "Northern America",
                    y: 161493,
                    drilldown: "Northern America"
                },
                {
                    name: "Central America",
                    y: 1190,
                    drilldown: "Central America"
                },
                {
                    name: "Caribbean Islands",
                    y: 445,
                    drilldown: "Caribbean Islands"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>

In [50]:
%%js
Highcharts.chart('good6', {
    chart: {
        type: 'bar',
        width: 300,
        height: 400
    },
    title: {
        text: 'Oceania'
    },
    subtitle: {
        text: 'This is the amount of Users that are located in different Coastel parts of Oceania'
    },
    xAxis: {
        categories: ['Australia and New Zealand','Polynesia','Micronesia','Melanesia',
                     
                     ],
        title: {
            text: null
        }
    },
    yAxis: {
        min: 0,
        title: {
            text: 'Location: Oceania | Users: 14,917',
            align: 'high'
        },
        labels: {
            overflow: 'justify'
        }
    },
    legend: { 
        enabled: false
    },
    tooltip: {
        valueSuffix: '%'
    },
    plotOptions: {
        bar: {
            dataLabels: {
                enabled: true
            }
        },
        series: {
            borderWidth: 0,
            dataLabels: {
                enabled: true,
                format: '{point.y:,0f}'
            }
        }
    },
    credits: {
        enabled: false
    },
    series: [
        {
            name: "Probability",
            color: "#e5b73b",
            data: [
                {
                    name: "Australia and New Zealand",
                    y: 14875,
                    drilldown: "Australia and New Zealand"
                },
                {
                    name: "Polynesia",
                    y: 10,
                    drilldown: "Polynesia"
                },
                {
                    name: "Micronesia",
                    y: 5,
                    drilldown: "Micronesia"
                },
                {
                    name: "Melanesia",
                    y: 27,
                    drilldown: "Melanesia"
                },
                
    
                
            ]
        }
    ]
});

<IPython.core.display.Javascript object>